File: source code/practice/src/main/java/com/decade/practice/web/security/UserClaims.java
package com.decade.practice.web.security;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.UUID;

@JsonIgnoreProperties(ignoreUnknown = true)
public record UserClaims(
          UUID id,
          String username,
          String name,
          String avatar
) {

}

File: practice/src/main/java/com/decade/practice/security/model/UserClaims.kt
package com.decade.practice.security.model

import com.decade.practice.model.entity.User
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import java.util.*

@JsonDeserialize
data class UserClaims(
      val id: UUID,
      val username: String,
      val name: String,
      val role: String,
      val gender: String,
) {
      constructor(user: User) : this(
            user.id,
            user.username,
            user.name,
            user.role,
            user.gender.random(),
      )
}